### ISSUE TITLE
Extend file source telemetry to include `processed_bytes_total`

### ISSUE THREAD

**Issue by blt** [timberio/vector#7973]
While working on #7955 I discovered that the file source does not report the bytes its processed. Compare to something like [stdin](https://vector.dev/docs/reference/configuration/sources/stdin/) source where `processed_bytes_total` is available, allowing us to calculate throughput for that source in terms of bytes per second. We would like to be able to make a bps claim for file source as well. 

Issue is resolved when file source emits telemetry sufficient to calculate a bytes per second throughput for the source.

**Comment by jszwedko** [comment]
👍 

I think I might like to see us continue the pattern of `events_in_total` and `events_out_total` and instead have two metrics: `bytes_in_total`, `bytes_out_total`. `bytes_in_total` would be reported by sources and `bytes_out_total` by sinks.

This is relevant to #3900

**Comment by spencergilbert** [comment]
> 👍
> 
> I think I might like to see us continue the pattern of `events_in_total` and `events_out_total` and instead have two metrics: `bytes_in_total`, `bytes_out_total`. `bytes_in_total` would be reported by sources and `bytes_out_total` by sinks.
> 
> This is relevant to #3900

Did you mean _not_ continue in the pattern of `events_in/out_total`?

**Comment by blt** [comment]
> +1
> 
> I think I might like to see us continue the pattern of `events_in_total` and `events_out_total` and instead have two metrics: `bytes_in_total`, `bytes_out_total`. `bytes_in_total` would be reported by sources and `bytes_out_total` by sinks.
> 
> This is relevant to #3900

Cool, so to confirm I understand you'd be interested in seeing us keep `events_{in,out}_total`, drop `processed_bytes_*` and add `bytes_{in,out}_total`? If so, I can get behind that.

**Comment by jszwedko** [comment]
> Cool, so to confirm I understand you'd be interested in seeing us keep `events_{in,out}_total`, drop `processed_bytes_*` and add `bytes_{in,out}_total`? If so, I can get behind that.

Exactly. Apologies, I can seem my original wording was not completely clear. I'd like to see us drop `processed_bytes_total` in-lieu of `bytes_{in,out}_total` to match up better with `events_in_total` and `events_out_total` which we replaced `processed_events_total` with.

**Comment by spencergilbert** [comment]
😆 I 100% agree with that. `{bytes/events}_{in/out}_total`

**Comment by blt** [comment]
Same.
